gemrb/core/GUI/TextSystem/TextContainer.cpp

```diff
@@ -366,33 +396,71 @@ Content* ContentContainer::RemoveContent(const Content* span, bool doLayout)
 		if (doLayout) {
 			LayoutContentsFrom(it);
 		}
+
+		ContentRemoved(content);
 		return content;
 	}
 	return NULL;
 }
 
+ContentContainer::ContentList::iterator
+ContentContainer::EraseContent(ContentList::iterator it)
+{
+	Content* content = *it;
+	content->parent = NULL;
+	layout.erase(std::find(layout.begin(), layout.end(), content));
+	layoutPoint = Point(); // reset cached layoutPoint
+	ContentRemoved(content);
+	delete content;
+
+	return contents.erase(it);
+}
+
+ContentContainer::ContentList::iterator
+ContentContainer::EraseContent(ContentList::iterator beg, ContentList::iterator end)
+{
+	for (; beg != end;) {
+		beg = EraseContent(beg);
+	}
+	return end;
+}
+
 Content* ContentContainer::ContentAtPoint(const Point& p) const
+{
+	const Layout* layout = LayoutAtPoint(p);
+	if (layout) {
+		// i know we are casting away const.
+		// we could return std::find(contents.begin(), contents.end(), *it) instead, but whats the point?
+		return (Content*)(layout->content);
+	}
+	return NULL;
+}
+
+const ContentContainer::Layout* ContentContainer::LayoutAtPoint(const Point& p) const
 {
 	// attempting to optimize the search by assuming content is evenly distributed vertically
 	// we are also assuming that layout regions are always contiguous and ordered ltr-ttb
 	// we do know by definition that the content itself is ordered ltr-ttb
 	// based on the above a simple binary search should suffice
 
+	int index = 0;
 	ContentLayout::const_iterator it = layout.begin();
 	size_t count = layout.size();
 	while (count > 0) {
 		size_t step = count / 2;
 		std::advance(it, step);
-		if ((*it).PointInside(p)) {
+		index += step;
```

Comment:
Codacy found an issue: [Variable 'index' is assigned a value that is never used.](https://app.codacy.com/manual/gemrb/gemrb/pullRequest?prid=6310305)